tests/test_ice_servers.py

```diff
@@ -0,0 +1,164 @@
+"""Test the ICE servers module."""
+
+import asyncio
+import time
+
+import pytest
+from webrtc_models import RTCIceServer
+
+from hass_nabucasa import ice_servers
+
+
+@pytest.fixture
+def ice_servers_api(auth_cloud_mock) -> ice_servers.IceServers:
+    """ICE servers API fixture."""
+    auth_cloud_mock.servicehandlers_server = "example.com/test"
+    auth_cloud_mock.id_token = "mock-id-token"
+    return ice_servers.IceServers(auth_cloud_mock)
+
+
+@pytest.fixture(autouse=True)
+def mock_ice_servers(aioclient_mock):
+    """Mock ICE servers."""
+    aioclient_mock.get(
+        "https://example.com/test/webrtc/ice_servers",
+        json=[
+            {
+                "urls": "turn:example.com:80",
+                "username": "12345678:test-user",
+                "credential": "secret-value",
+            },
+        ],
+    )
+
+
+async def test_ice_servers_listener_registration_triggers_periodic_ice_servers_update(
+    ice_servers_api: ice_servers.IceServers,
+):
+    """Test that registering an ICE servers listener triggers a periodic update."""
+    times_register_called_successfully = 0
+
+    ice_servers_api._get_refresh_sleep_time = lambda: -1
+
+    async def register_ice_servers(ice_servers: list[RTCIceServer]):
+        nonlocal times_register_called_successfully
+
+        # There asserts will silently fail and variable will not be incremented
```

Comment:
As this is a callback, the test runner does not detect it failing in error logs; it just stops the further execution of the test. This could be due to try/catch in the code.

Reply:
Right, that's just normal test behavior when using assert. I'd not add a comment about that. It's confusing to me.

Reply:
I think I understand now. This is happening in a separate task, not the same as the test, and that's why the test output is different. Is that correct?

Reply:
Yes, that is probably the reason.